src/components/results/securityanalysis/security-analysis-result-tab.tsx

```diff
@@ -5,76 +5,157 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import { FunctionComponent } from 'react';
+import React, {
+    SyntheticEvent,
+    FunctionComponent,
+    useEffect,
+    useState,
+} from 'react';
+import { useSelector } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { ReduxState } from '../../../redux/reducer.type';
+import { Box } from '@mui/system';
+import { Tabs, Tab, Select, MenuItem, LinearProgress } from '@mui/material';
+import { fetchSecurityAnalysisResult } from '../../../services/study/security-analysis';
+import { useOpenLoaderShortWait } from '../../dialogs/commons/handle-loader';
+import { RunningStatus } from '../../utils/running-status';
+import { RESULTS_LOADING_DELAY } from '../../network/constants';
+import { ComputingType } from '../../computing-status/computing-type';
+import { SecurityAnalysisResultN } from './security-analysis-result-n';
+import { SecurityAnalysisResultNmk } from './security-analysis-result-nmk';
 import {
-    NmKConstraintRow,
+    SecurityAnalysisResultType,
     SecurityAnalysisTabProps,
 } from './security-analysis.type';
-import { useNodeData } from '../../study-container';
-import { fetchSecurityAnalysisResult } from '../../../services/study/security-analysis';
-import { SecurityAnalysisResult } from './security-analysis-result';
-import { ColDef } from 'ag-grid-community/dist/lib/entities/colDef';
-import { useSnackMessage } from '@gridsuite/commons-ui';
-import { fetchLineOrTransformer } from '../../../services/study/network-map';
+import { NMK_TYPE, RESULT_TYPE } from './security-analysis-result-utils';
+
+const styles = {
+    container: {
+        display: 'flex',
+        position: 'relative',
+    },
+    tabs: {
+        position: 'relative',
+        top: 0,
+        left: 0,
+    },
+    nmkResultSelect: {
+        position: 'absolute',
+        right: '16px',
+    },
+    loader: {
+        height: '4px',
+    },
+    resultContainer: {
+        flexGrow: 1,
+    },
+};
 
 export const SecurityAnalysisResultTab: FunctionComponent<
     SecurityAnalysisTabProps
 > = ({ studyUuid, nodeUuid, openVoltageLevelDiagram }) => {
-    const securityAnalysisResultInvalidations = ['securityAnalysisResult'];
-    const { snackError } = useSnackMessage();
+    const [tabIndex, setTabIndex] = useState(0);
+    const [nmkType, setNmkType] = useState(
+        NMK_TYPE.CONSTRAINTS_FROM_CONTINGENCIES
+    );
+    const [result, setResult] = useState<SecurityAnalysisResultType>(null);
+    const [isLoadingResult, setIsLoadingResult] = useState(false);
 
-    const [securityAnalysisResult, isWaiting] = useNodeData(
```

Comment:
Ditching useNodeData to manage the results may has the downside of not updating the results if you are already in the result tab while executing the computation, you have to navigate or refresh the page to display the results. I think it's because `securityAnalysisResult` is no longer used to invalidate and subsequently fetching the results.

Reply:
done, thnak you